rustual-boy-core/src/rom.rs

```diff
@@ -22,12 +22,15 @@ impl Rom {
         let mut vec = Vec::new();
         file.read_to_end(&mut vec)?;
 
-        let size = vec.len();
+        Rom::from_bytes(vec.into_boxed_slice())
+    }
+
+    pub fn from_bytes(mut bytes: Box<[u8]>) -> io::Result<Rom> {
```

Comment:
Perhaps taking a `&[u8]` would be better than a `Box<[u8]>` as it's more general (basically all sequences in Rust coerce to this type) and more idiomatic (for that reason). Other than that this looks great, and I agree the file I/O is probably better left to the frontend eventually.